core/trino-spi/src/main/java/io/trino/spi/connector/ColumnsMetadata.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class ColumnsMetadata
```

Comment:
`io.trino.spi.connector.ConnectorTableSchema`?

Reply:
Ultimately we need `ColumnMetadata` objects from `MetadataListing#listTableColumns`, which seems to more information than `ColumnSchema` in `ConnectorTableSchema` right now. Since the purpose of `ConnectorTableSchema` seems to be avoiding fetching full metadata, may be we can keep the two separate. wdyt? 

also open to comments/suggestions for class/method naming here. May be we can use `Stream<TableColumnsMetadata> streamTableColumns(...)` instead of current `Stream<ColumnsMetadata> listTableColumnsStream(...)`.